crates/symbolicator/src/config.rs

```diff
@@ -420,6 +420,44 @@ mod tests {
         )
     }
 
+    #[test]
+    fn test_unspecified_dl_timeouts() {
+        let yaml = r#"
+            sources: []
+        "#;
+        let cfg = Config::from_reader(yaml.as_bytes()).unwrap();
+        let default_cfg = Config::default();
+        assert_eq!(cfg.max_download_timeout, default_cfg.max_download_timeout);
+        assert_eq!(cfg.connect_timeout, default_cfg.connect_timeout);
+        assert_eq!(cfg.streaming_timeout, default_cfg.streaming_timeout);
+    }
+
+    #[test]
+    fn test_zero_second_dl_timeouts() {
+        // 0s download timeouts will not be set to defaults
+        let yaml = r#"
+            max_download_timeout: 0s
+            connect_timeout: 0s
+            streaming_timeout: 0s
+        "#;
+        let cfg = Config::from_reader(yaml.as_bytes()).unwrap();
+        assert_eq!(cfg.max_download_timeout, Duration::from_secs(0));
+        assert_eq!(cfg.connect_timeout, Duration::from_secs(0));
+        assert_eq!(cfg.streaming_timeout, Duration::from_secs(0));
+    }
+
+    #[test]
+    fn test_null_dl_timeouts() {
```

Comment:
Should these timeouts actually be optional like the cache timeouts in case somebody does want disable download timeouts? If not, I can remove this test. If we do want them, I can open up a follow-up PR that adds that functionality in.